symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -197,60 +190,75 @@ public V4Message send(@Nonnull V4Stream stream, @Nonnull String message) {
-          message.getData(),
-          message.getVersion(),
-          tempFile,
-          null
-          )
-      );
-    } finally {
-      try {
-        Files.deleteIfExists(tempFile.toPath());
-      } catch (IOException ignored) {
-
-      }
-    }
+  public V4Message send(@Nonnull V4Stream stream, @Nonnull Message message) {
+    return this.send(stream.getStreamId(), message);
   }
 
   /**
    * Sends a message to the stream ID passed in parameter.
    *
-   * @param stream    the stream to send the message to
-   * @param message   the message to send to the stream
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
-  public V4Message send(@Nonnull V4Stream stream, @Nonnull Message message) {
-    return send(stream.getStreamId(), message);
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
+    return this.executeAndRetry("send", () ->
+        this.doSend(streamId, message, this.authSession.getSessionToken(), this.authSession.getKeyManagerToken())
+    );
+  }
+
+  /**
+   * The generated {@link MessagesApi#v4StreamSidMessageCreatePost(String, String, String, String, String, String, File, File)}
+   * does not allow to send multiple attachments as well as in-memory files, so we have to "manually" process this call.
+   */
+  private V4Message doSend(String streamId, Message message, String sessionToken, String keyManagerToken) throws ApiException {
+    final ApiClient apiClient = this.messagesApi.getApiClient();
+    final Map<String, Object> form = new HashMap<>();
+    form.put("message", message.getContent());
+    form.put("data", message.getData());
+    form.put("version", message.getVersion());
+    form.put("attachment", toApiClientBodyParts(message.getAttachments()));
+    form.put("preview", toApiClientBodyParts(message.getPreviews()));
+
+    final Map<String, String> headers = new HashMap<>();
+    headers.put("sessionToken", apiClient.parameterToString(sessionToken));
+    headers.put("keyManagerToken", apiClient.parameterToString(keyManagerToken));
+
+    return apiClient.invokeAPI(
+        "/v4/stream/" + apiClient.escapeString(streamId) + "/message/create",
+        "POST",
+        emptyList(),
+        null, // for 'multipart/form-data', body can be null
+        headers,
+        emptyMap(),
+        form,
+        apiClient.selectHeaderAccept("application/json"),
+        apiClient.selectHeaderContentType("multipart/form-data"),
+        new String[0],
+        new TypeReference<V4Message>() {}
+    ).getData();
+  }
+
+  private static ApiClientBodyPart[] toApiClientBodyParts(List<Attachment> attachments) {
+    final ApiClientBodyPart[] result = new ApiClientBodyPart[attachments.size()];
```

Comment:
could probably be written with streams (just to avoid the index iteration on the attachments)